SingularityS3Uploader/src/main/java/com/hubspot/singularity/s3uploader/SingularityS3Uploader.java

```diff
@@ -148,19 +150,23 @@ private void uploadBatch(List<Path> toUpload) {
     for (int i = 0; i < toUpload.size(); i++) {
       final Context context = metrics.getUploadTimer().time();
       final Path file = toUpload.get(i);
-      try {
-        uploadSingle(i, file);
-        metrics.upload();
-        success++;
-        Files.delete(file);
-      } catch (S3ServiceException se) {
-        metrics.error();
-        LOG.warn("{} Couldn't upload {} due to {} ({}) - {}", logIdentifier, file, se.getErrorCode(), se.getResponseCode(), se.getErrorMessage(), se);
-      } catch (Exception e) {
-        metrics.error();
-        LOG.warn("{} Couldn't upload or delete {}", logIdentifier, file, e);
-      } finally {
-        context.stop();
+      if (!fileOpen(file)) {
```

Comment:
chatted about this in person, but let's make this configurable in the S3 uploader config file